src/axom/inlet/tests/inlet_Inlet.cpp

```diff
@@ -1058,8 +1058,119 @@ TEST(inlet_Inlet_verify, verifyValidStringValues) {
+}
+
+TEST(inlet_verify, verifyFieldLambda) {
+  std::string testString = "field1 = true; field2 = 'abc'; NewTable = { field3 = 'xyz'; field4 = 'yes' }";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  auto field1 = inlet->addBool("field1");
+  auto field2 = inlet->addString("field2");
+  auto field3 = inlet->addString("NewTable/field3");
+
+  field2->registerVerifier([&]() -> bool {
+    std::string str = "";
+    inlet->get("field2", str);
+    return (str.size() >= 1 && str[0] == 'a');
+  });
+  EXPECT_TRUE(inlet->verify());
+
+  field3->registerVerifier([&]() -> bool {
+    std::string str = "";
+    inlet->get("NewTable/field3", str);
+    return (str.size() >= 1 && str[0] == 'a');
+  });
+  EXPECT_FALSE(inlet->verify());
+}
+
+TEST(inlet_verify, verifyTableLambda) {
+  std::string testString = "field1 = true; field2 = 'abc'; NewTable = { field3 = 'xyz'; field4 = 'yes' }";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  auto field1 = inlet->addBool("field1");
+  auto field2 = inlet->addString("field2");
+  auto table1 = inlet->addTable("NewTable");
+  auto field3 = table1->addString("field3");
+
+  field2->registerVerifier([&]() -> bool {
+    std::string str = "";
+    inlet->get("field2", str);
+    return (str.size() >= 1 && str[0] == 'a');
+  });
+  EXPECT_TRUE(inlet->verify());
+
+  field3->registerVerifier([&]() -> bool {
+    std::string str = "";
+    inlet->get("NewTable/field3", str);
+    return (str.size() >= 1 && str[0] == 'x');
+  });
+  EXPECT_TRUE(inlet->verify());
+
+  table1->registerVerifier([&]() -> bool {
+    return table1->hasField("field3");
+  });
+  EXPECT_TRUE(inlet->verify());
+
+  table1->registerVerifier([&]() -> bool {
+    return table1->hasField("field22");
+  });
+  EXPECT_FALSE(inlet->verify());
 
+  std::string testString2 = "thermal_solver={}\n"
+                            "thermal_solver.order = 2\n"
+                            "thermal_solver.timestepper = 'quasistatic'\n"
+                            "solid_solver={}\n"
+                            "solid_solver.order = 3\n"
+                            "solid_solver.timestepper = 'BackwardEuler'\n"
+                            "material={}\n"
+                            "material.attribute = 1\n"
+                            "material.thermalview = 'isotropic'\n"
+                            "material.solidview = 'hyperelastic'";
+  DataStore ds2;
+  auto inlet2 = createBasicInlet(&ds2, testString2);
+  auto thermalOrder = inlet2->addInt("thermal_solver/order");
+  auto thermalTimeStep = inlet2->addString("thermal_solver/timestepper");
+  auto solidOrder = inlet2->addInt("solid_solver/order");
+  auto soldTimestep = inlet2->addString("solid_solver/timestepper");
+  auto attrib = inlet2->addInt("material/attribute");
+  auto globalTable = inlet2->getGlobalTable();
+  auto material = globalTable->getTable("material");
+
+  globalTable->registerVerifier([&]() -> bool {
```

Comment:
I haven't gotten there yet, but does inlet support multiple verifiers on the same Field or Table?

Reply:
In either case, perhaps there should be a test that attempts to register more than one verifier.

Reply:
It does not support more than one verifier per Field or Table, it should throw a warning if that is the case and just overwrite what was there.

I agree there should be a test.